server/src/main/java/org/opensearch/indices/recovery/MultiFileWriter.java

```diff
@@ -128,6 +131,22 @@ public IndexOutput openAndPutIndexOutput(String fileName, StoreFileMetadata meta
         return indexOutput;
     }
 
+    /**
+     * Helper method that determines if we should fsync temporary file being copied by MultiFileWriter. In the case of segment replication,
+     * we only want to fsync files that are part of a greater commit point while we fsync all files in the case of document replication.
+     * @return boolean - true if we should fsync temporary file; false if not
+     */
+    private boolean shouldFsync() {
```

Comment:
This makes sense, but rather than computing it for every file I think we can precompute it in this class's constructor and store it as a field.

Reply:
I tried that initially but at the time MFW's ctor is called, there are no files added to fileDetails so it always returns false.

Reply:
Ah, what about adding a field to FileDetails that flips as addDoc is called?